backend/ml/management/commands/ml_train.py

```diff
@@ -100,6 +101,14 @@ def save_data(video_scores, contributor_rating_scores, poll, trusted_only=True):
             for video_id, criteria, score, uncertainty in video_scores
         )
 
+        entities = []
+        for entity in Entity.objects.prefetch_related("criteria_scores").iterator():
+            entity.tournesol_score = 10 * sum(
+                criterion.score for criterion in entity.criteria_scores.iterator()
+            )
+            entities.append(entity)
+        Entity.objects.bulk_update(entities, ["tournesol_score"])
```

Comment:
I see some issues with this implementation: 
 * all entities will be updated, including entities not related to the current poll, as well as entities without score (where `tournesol_score` should remain null)
 * in Django 4.0 [.prefech_related() is ignored when used with `.iterator()`](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#prefetch-related`.iterator)
 Although it may change in future releases, see https://github.com/django/django/pull/15334)

My suggestion would be to remove `.iterator()` and filter on the score poll;
```python
        for entity in (
            Entity.objects.filter(criteria_scores__poll=poll)
            .distinct()
            .prefetch_related("criteria_scores")
        ):
```

Reply:
I find it weird to fetch the criteria scores from the db in order to compute the tournesol_score at this moment because the ML algorithm just finished computing them and we are in the saving phase. Why not computing the tournesol_scores directly from the list `video_scores` returned by the ML algorithm?

Also we may in the fuure make the tournesol_score more complicated thant just `10 x sum(criteria_scores)`. It would probably be beneficial to already separate the computation in its own method.

Reply:
> I find it weird to fetch the criteria scores from the db in order to compute the tournesol_score at this moment because the ML algorithm just finished computing them and we are in the saving phase. Why not computing the tournesol_scores directly from the list video_scores returned by the ML algorithm?

I've considered it at first, but the loop would have been less readable and more bug-prone, mainly because the items in `video_scores` are not entities, but criteria with their score ordered by criteria name. It would have required either a a copy with a transformation of the data structure, or a loop with several conditions inside. That's why I've opted for an extra SQL query, which simplify all of this with a rather low cost.

> Also we may in the future make the tournesol_score more complicated thant just 10 x sum(criteria_scores). It would probably be beneficial to already separate the computation in its own method.

That's a good idea. As the algorithm is going to change I don't know if it's still relevant but yes, it's a good idea.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/614/commits/a48fd5d4ca5e65b8714b5314da3ee41e0080eaa1.